fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -4,10 +4,66 @@
 # LICENSE file in the root directory of this source tree.
 
 from enum import Enum
+from typing import Optional
 
+import torch
 from torch import Tensor
 
 
+# Helper Functions
+def get_k_for_topk(
+    tensor: Tensor, topk_percent: Optional[float], top_k_element: Optional[int], topk_dim: Optional[int]
+) -> int:
+    """Converts the top_k_percent to top_k_element when top_k_percent is provided
+    as the criterion for top-k calculation. When, top_k_element is used as the criterion,
+    simply returns the value for k. Also, ensures k is never 0 to avoid all sparse tensors.
+
+    Args:
+        tensor (Tensor):
+            Input tensor.
+        topk_percent (float, optional)
+            Percent of top-k elements to retain for SST. Default: 0.1
+
+    Returns:
+        k (int)
+            the value of k for use as top-k
+    """
+    if top_k_element is None:
+        dim_numel = tensor.numel() if topk_dim is None else tensor.shape[topk_dim]
+        k = int(dim_numel * topk_percent / 100.0)
+    else:
+        k = top_k_element
+
+    # ensure we never have 100% sparsity in tensor and always have 1 surviving element!
+    k = k if k != 0 else 1
+    return k
+
+
+def scatter_topk_to_sparse_tensor(src_tensor: Tensor, dest_tensor: Tensor, k: int, dim: Optional[int]) -> Tensor:
+    """Scatter the topk values of the dest_tensor to a zero tensor of the same shape
```

Comment:
let's rename src_tensor -> top_k_tensor, dest_tensor -> to_be_sparsify_tensor.

should comment why we are doing this. The reason is that the top-k computation might be done with a derived tensor from to_be_sparsified_tensor.

Also, assert these 2 tensors have the same shapes.

Reply:
naming is very important for understanding and cognitive load of the code.